Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1078,6 +1118,40 @@ bool LineBuilder::tryPlacingFloatBox(const InlineItem& floatItem, LineBoxConstra
     auto& floatBox = floatItem.layoutBox();
     ASSERT(formattingState());
     auto& boxGeometry = formattingState()->boxGeometry(floatBox);
+    auto containingBlockMarginTrim = floatBox.parent().style().marginTrim();
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    auto trimMarginsForPositionedFloat = [&](LayoutPoint& floatPosition) {
```

Comment:
tryPlacingFloatBox is already a rather long function and trimming is not a core(common) functionality. Please move this section to a local static helper to keep tryPlacingFloatBox "short and clean".